crispr_read_counts/utils.py

```diff
@@ -43,25 +44,38 @@ def open_plain_or_gzipped_file(file: str):
         f.close()
 
 
-def check_file_readable(fn):
+def check_file_readable(fn, msg_if_fail=None):
+  result = True
   if os.path.isfile(fn):
-    return os.access(fn, os.R_OK)
+    result = os.access(fn, os.R_OK)
   else:
-    return False
+    result = False
+
+  if not result:
+    if msg_if_fail:
+      sys.exit(error_msg(msg_if_fail))
+    else:
+      sys.exit()
 
 
-def check_file_writable(fn):
-  '''the function is clearly borrowed from web'''
+def check_file_writable(fn, msg_if_fail=None):
+  result = True
   if os.path.exists(fn):
     # path exists
     if os.path.isfile(fn):  # is it a file or a dir?
       # also works when file is a link and the target is writable
-      return os.access(fn, os.W_OK)
+      result = os.access(fn, os.W_OK)
+    else:
+      result = False  # path is a dir, so cannot write as a file
+  else:  # target does not exist, check perms on parent dir
+    pdir = os.path.dirname(fn)
+    if not pdir:
+      pdir = '.'
+    # target is creatable if parent dir is writable
+    result = os.access(pdir, os.W_OK)
```

Comment:
This should be equivalent:

```python
result: bool = (
  (os.path.isfile(fn) and os.access(fn, os.W_OK)) if os.path.exists(fn) else
  os.access(os.path.dirname(fn) or '.', os.W_OK)
)
```

If the user mistypes the file name the check will fall back to evaluating the directory, though. Is that okay?

Reply:
I've been thinking about what those lines can be :)

I thought those "if else" looked a bit clearer, so what's the selling point of your line?

Reply:
The nested conditional blocks could perform any action, so you have to read through all of them to understand what they affect; an expression can only return a value, so the purpose is clearer at first glance.

Reply:
You're right. The point is taken :)